eth/db/account.py

```diff
@@ -533,12 +533,12 @@ def _get_meta_witness(self) -> MetaWitness:
 
     def _validate_generated_root(self) -> None:
         db_diff = self._journaldb.diff()
-        if len(db_diff):
+        if db_diff:
```

Comment:
just noting that we need to be sure `db_diff` implements a `__bool__` otherwise this will return `True` unconditionally because of python's object model.

Reply:
Looks like we're inheriting from `Mapping`

https://github.com/ethereum/py-evm/blob/9e6749fe6083e47b0522dd479ffcff7ee3214391/eth/db/diff.py#L104

Which suggests to me this is :+1:

Reply:
Do you want me to revert only this occurrence, or everywhere where `len()` was replaced?

Reply:
Don't need anything reverted.  This change is fine and the other removals of `len` appeared to be on data structures that are fine to do direct boolean checks.